sumeh/engines/pandas/analyzers.py

```
@@ -0,0 +1,521 @@
+# ============================================================================
+
+
+class MembershipAnalyzer:
+    @staticmethod
+    def analyze(df: pd.DataFrame, rule: RuleDefinition) -> MetricResult:
+        field = rule.field
+        allowed_values = rule.value
+        check_type = rule.check_type
+
+        if field not in df.columns:
+            raise KeyError(f"Field '{field}' not found")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)